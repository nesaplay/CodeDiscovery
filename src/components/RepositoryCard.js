import React from 'react'
import { Card, Icon, Label } from 'semantic-ui-react'
import { Link } from 'react-router-dom'

const RepositoryCard = ({
	full_name,
	description,
	language,
	html_url,
	stargazers_count,
	open_issues_count,
	contributors_url,
}) => {
	const { Content, Header, Meta, Description } = Card

	return (
		<Card fluid color='orange'>
			<Content>
				<Link to={`/repos/${full_name}/contributors`}>
					<Label color="orange" ribbon="right">
						Top Contributors
					</Label>
				</Link>
				<Header>
					<a href={html_url} target="_blank">
						{full_name}
					</a>
				</Header>
				<Meta>{language}</Meta>
				<Description>{description}</Description>
			</Content>
			<Content extra>
				<span className='ui left floated'>
					<Icon name="star" /> {stargazers_count}
				</span>
				<span className='ui right floated'>
					<Icon name="warning sign" /> {open_issues_count}
				</span>
			</Content>
		</Card>
	)
}

export default RepositoryCard
