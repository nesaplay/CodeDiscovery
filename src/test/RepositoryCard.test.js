import React from 'react'
import RepositoryCard from '../components/RepositoryCard'

it('renders a component', () => {
	const wrapper = shallow(
        <RepositoryCard />
    )

	expect(wrapper).toMatchSnapshot()
})
