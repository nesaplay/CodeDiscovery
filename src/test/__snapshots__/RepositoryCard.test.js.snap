// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a component 1`] = `
<Card
  color="orange"
  fluid={true}
>
  <CardContent>
    <Link
      replace={false}
      to="/repos/undefined/contributors"
    >
      <Label
        color="orange"
        ribbon="right"
      >
        Top Contributors
      </Label>
    </Link>
    <CardHeader>
      <a
        target="_blank"
      />
    </CardHeader>
    <CardMeta />
    <CardDescription />
  </CardContent>
  <CardContent
    extra={true}
  >
    <span
      className="ui left floated"
    >
      <Icon
        as="i"
        name="star"
      />
       
    </span>
    <span
      className="ui right floated"
    >
      <Icon
        as="i"
        name="warning sign"
      />
       
    </span>
  </CardContent>
</Card>
`;
